src/3box.js

```diff
@@ -378,6 +346,15 @@ class Box {
     }
   }
 
+  async _ensurePinningNodeConnected (odbAddress) {
+    const roomPeers = await this._ipfs.pubsub.peers(odbAddress)
+    if (!roomPeers.find(p => p === this.pinningNode.split('/').pop())) {
+      this._ipfs.swarm.connect(this.pinningNode)
+      const rootStoreAddress = this._rootStore.address.toString()
+      this._pubsub.publish(PINNING_ROOM, { type: 'PIN_DB', odbAddress: rootStoreAddress })
```

Comment:
Does this not have to wait for the peer (pinning node to be connected) before pushing this message? And should any of this block/await before setting or removing?

Reply:
oh well i guess it doesnt have to block set or remove, since it will eventually share those updates once connected

Reply:
Yeah, there are two cases:
1. We lost the connection to the pinning node. Then the onPeer function will send the msg to the pinning node once connected.
2. We are connected to the pinning node, but it doesn't have our db open. Then we just publish to the pinning room.

Reply:
Initially I had separate logic for bot cases, but that was a bit redundant I think.